### PR TITLE
feat: Add secure MessagBus capability

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-mqtt-go/blob/master/.github/CONTRIBUTING.md

## What is the current behavior?
no secure MessageBus capability

## What is the new behavior?
Secure MessageBus capability added

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: cmd/res/configuration.toml [hunk 1]
```diff
@@ -38,11 +38,12 @@ Type = 'consul'
   Port = 48081
 
 [MessageQueue]
-Enabled = false
-Protocol = 'tcp'
+Protocol = 'redis'
```
Comment (hahattan) [inline cmd/res/configuration.toml:41]: miss `UseMessageBus = false` under `Service` section

**Review by hahattan — CHANGES_REQUESTED** [review]
Should we also upgrade device service to Go 1.16 as SDK is using it?

File: cmd/res/configuration.toml (hunk 1 above)
Comment (lenny-intel) [reply cmd/res/configuration.toml:41]: Humm, I remember double checking that... :-( Good catch!

**Comment by hahattan** [comment]
> > I think we should also upgrade device service to use Go1.16
> 
> Yes, but needs to be in a separate PR.

Might want to downgrade sdk to `-.dev.52` otherwise build would fail in this scenario?

**Review by cloudxxx8 — APPROVED** [review]
LGTM
@hahattan please help review again

**Review by hahattan — APPROVED** [review]
LGTM

**Pull request merged**